DotNetHowToIndexers/Program.cs

```diff
@@ -0,0 +1,325 @@
+            {
+                indexerName = SetupAzureTableStorageIndexer(serviceClient, configuration);
+            }
+            else if (option.Equals(AzureCosmosDbOption, StringComparison.OrdinalIgnoreCase))
+            {
+                indexerName = SetupAzureCosmosDbIndexer(serviceClient, configuration);
+            }
+            else
+            {
+                Usage();
+                return;
+            }
+
+            Console.WriteLine("Deleting index...");
+            DeleteIndexIfExists(serviceClient);
+
+            Console.WriteLine("Creating index...");
+            CreateIndex(serviceClient);
+
+            Console.WriteLine("Running indexer...");
+            serviceClient.Indexers.Run(indexerName);
+
+            Console.WriteLine("Done...");
+            Console.ReadKey();
+        }
+
+        /// <summary>
+        /// Creates an indexer which pulls data from an Azure SQL database and puts it
+        /// inside the hotels index.
+        /// </summary>
+        private static string SetupAzureSqlIndexer(SearchServiceClient serviceClient, IConfigurationRoot configuration)
+        {
+            Console.WriteLine("Deleting Azure SQL data source if it exists...");
+            DeleteDataSourceIfExists(serviceClient, AzureSqlDataSourceName);
+
+            Console.WriteLine("Creating Azure SQL data source...");
+            DataSource azureSqlDataSource = CreateAzureSqlDataSource(serviceClient, configuration);
+
+            Console.WriteLine("Deleting Azure SQL indexer if it exists...");
+            DeleteIndexerIfExists(serviceClient, AzureSqlIndexerName);
+
+            Console.WriteLine("Creating Azure SQL indexer...");
+            Indexer azureSqlIndexer = CreateIndexer(serviceClient, AzureSqlDataSourceName, AzureSqlIndexerName);
+
+            return azureSqlIndexer.Name;
+        }
+
+        /// <summary>
+        /// Creates an indexer which pulls data from Azure Table Storage and puts it
+        /// inside the hotels index.
+        /// </summary>
+        private static string SetupAzureTableStorageIndexer(SearchServiceClient serviceClient, IConfigurationRoot configuration)
+        {
+            Console.WriteLine("Deleting Azure Table Storage data source if it exists...");
+            DeleteDataSourceIfExists(serviceClient, AzureTableStorageDataSourceName);
+
+            Console.WriteLine("Creating Azure Table Storage data source...");
+            DataSource azureTableStorageDataSource = CreateAzureTableStorageDataSource(serviceClient, configuration);
+
+            Console.WriteLine("Deleting Azure Table Storage indexer...");
+            DeleteIndexerIfExists(serviceClient, AzureTableStorageIndexerName);
+
+            Console.WriteLine("Creating Azure Table Storage indexer...");
+            Indexer azureTableStorageIndexer = CreateIndexer(serviceClient, AzureTableStorageDataSourceName, AzureTableStorageIndexerName);
+
+            return azureTableStorageIndexer.Name;
```

Comment:
General point around this pattern... I wonder if instead of using the _DeleteIfExists_, _Create_ pattern, could we simply use the _CreateOrUpdate_ pattern? Is there any benefit to not leveraging the update semantics?

Reply:
This pattern probably started with indexes. Sometimes users will extend those in ways that make updating them later impossible (for example, they might add new fields that aren't in the sample code, causing the service to reject the update request because it thinks they're trying to delete a field). Deleting and then re-creating the index ensures that the sample starts from a clean slate.

All that said, indexer and datasource updates aren't restricted in the same way, so we could maybe use update, but then we'd have two ways of doing things in the same sample. I'm not very opinionated on this either way, but I figured it would help to provide some context.

Reply:
I'll leave this alone for now.

Reply:
@brjohnstmsft Thanks for the explanation.. I don't mind the pattern at all, was just curious of the context of using it, which I have now.